### PR TITLE
chore(multiple): localize powertools_dev env logic and warning

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #1527

Documentation being updated in https://github.com/awslabs/aws-lambda-powertools-python/pull/1569

### Changes

> Please provide a summary of what's being changed

* Localize logic for the new `POWERTOOLS_DEV` to ease adopting in other utilities later
* Add support for `POWERTOOLS_DEV` in Event Handler

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: aws_lambda_powertools/event_handler/api_gateway.py
Comment (leandrodamascena) [inline aws_lambda_powertools/event_handler/api_gateway.py:538]: We need to update the [documentation](https://awslabs.github.io/aws-lambda-powertools-python/1.29.2/core/event_handler/api_gateway/#debug-mode). #1569 

We have 2 options and in my opinion we should replace the environment variable name in the documentation.

Adding a warning
image

Replacing
image

File: aws_lambda_powertools/shared/functions.py
Comment (leandrodamascena) [inline aws_lambda_powertools/shared/functions.py:7]:
```suggestion
import base64
import logging
import os
import warnings
from binascii import Error as BinAsciiError
from typing import Optional, Union

from aws_lambda_powertools.shared import constants
```

File: aws_lambda_powertools/shared/functions.py
Comment (leandrodamascena) [inline aws_lambda_powertools/shared/functions.py]:
```suggestion
    is_on = strtobool(os.getenv(constants.POWERTOOLS_DEV_ENV, "0"))
```

**Review by leandrodamascena — DISMISSED** [review]
I made two suggestions. I executed it locally and in Lambda and it worked as expected.
Please check if it make sense.

File: aws_lambda_powertools/event_handler/api_gateway.py
Comment (heitorlessa) [reply aws_lambda_powertools/event_handler/api_gateway.py:538]: Good catch, let's replace it. I think you meant to make a comment in the [docs PR](https://github.com/awslabs/aws-lambda-powertools-python/pull/1569) - I'll assume you approve the docs PR after this change, then I'll merge.

Comment (heitorlessa) [reply aws_lambda_powertools/event_handler/api_gateway.py:538]: done - resolving.

File: aws_lambda_powertools/shared/functions.py
Comment (heitorlessa) [reply aws_lambda_powertools/shared/functions.py:7]: Discarding this suggestion as this PR passes isort checks

File: aws_lambda_powertools/shared/functions.py
Comment (heitorlessa) [reply aws_lambda_powertools/shared/functions.py]: great catch!

**Comment by heitorlessa** [comment]
> I made two suggestions. I executed it locally and in Lambda and it worked as expected. Please check if it make sense.

Thanks a lot for the useful suggestions!

**Pull request merged**
